fiat/fiat-github/src/test/java/com/netflix/spinnaker/fiat/config/GitHubConfigTest.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2025 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.fiat.config;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import com.netflix.spinnaker.config.OkHttp3ClientConfiguration;
+import com.netflix.spinnaker.fiat.roles.github.GitHubProperties;
+import com.netflix.spinnaker.fiat.roles.github.client.GitHubClient;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyPairGenerator;
+import java.security.interfaces.RSAPrivateKey;
+import java.util.Base64;
+import okhttp3.OkHttpClient;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class GitHubConfigTest {
+
+  @Mock private OkHttp3ClientConfiguration mockOkHttpClientConfig;
+
+  private GitHubConfig gitHubConfig;
+  private GitHubProperties gitHubProperties;
+  private Path tempPrivateKeyFile;
+  private OkHttpClient.Builder realClientBuilder;
+
+  @BeforeEach
+  void setUp(@TempDir Path tempDir) throws Exception {
+    gitHubConfig = new GitHubConfig();
+    gitHubProperties = new GitHubProperties();
+    gitHubProperties.setBaseUrl("https://api.github.com");
+    gitHubProperties.setOrganization("test-org");
+    gitHubConfig.setGitHubProperties(gitHubProperties);
+
+    // Create test private key
+    KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
```

Comment:
So this would work but I kinda wonder about just generating a VERY long lived key instead of creating the key every test.  This would reduce a fair bit of complexity.

Reply:
*ponder* I've generated this in the past, it's not END Of the world ;) Just simplifies that you're testing key decoding vs. key creation.  I'd be more ok if we were testing standard key handling in GENERIC.  This isn't a "no" per se, but... more of "hrmm" moment.